File: shared-client-libs/src/main/java/eu/getsoftware/hotelico/clients/api/application/dto/entity/CustomerDealDTO.java
package eu.getsoftware.hotelico.clients.api.application.dto.entity;

import eu.getsoftware.hotelico.clients.common.domain.ids.CustomerDealDomainEntityId;
import eu.getsoftware.hotelico.clients.common.domain.ids.CustomerDomainEntityId;
import eu.getsoftware.hotelico.clients.common.domain.ids.HotelDomainEntityId;
import eu.getsoftware.hotelico.clients.common.dto.BasicDTO;
import eu.getsoftware.hotelico.clients.common.utils.AppConfigProperties;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.Date;

/**
 * DTO lieber immutable - all field are final and no side effects!
 */
@Getter
//@Builder
@RequiredArgsConstructor
public class CustomerDealDTO implements BasicDTO<CustomerDealDomainEntityId> {
    
    private final CustomerDealDomainEntityId domainEntityId;
    private final boolean active = true;

    private long activityId = 0L;
    private long consistencyId = 0L;

    private boolean timeValid = true;
    private boolean lastMinute = false;
    private boolean closed = false;

    private HotelDomainEntityId hotelDomainId;
    private CustomerDomainEntityId senderDomainId;

    private int dealDaysDuration = 0;
    private Double totalMoney;

    private Date validFrom;
    private Date validTo;

    private String title = "";
    private String timeString = "";

    private String activityArea = "";
    private String dealCode = "";
    private String tablePosition = "";
    private String dealStatus = "";
    private String shortDescription = "";
    private String description = "";

    private String pictureUrl;
    private String previewPictureUrl;

    public String getValidToString() {
        return validTo != null ? AppConfigProperties.dateFormat.format(validTo) : "";
    }

    public String getValidFromString() {
        return validFrom != null ? AppConfigProperties.dateFormat.format(validFrom) : "";
    }

    public String getPictureUrl() {
        return AppConfigProperties.addHostPrefixOnDemand(pictureUrl);
    }

    public String getPreviewPictureUrl() {
        String picUrl = AppConfigProperties.isEmptyString(previewPictureUrl)
                ? AppConfigProperties.PREVIEW_HOTEL_NOT_AVAILABLE_URL
                : previewPictureUrl;

        return AppConfigProperties.addHostPrefixOnDemand(picUrl);
    }

    public String getTimeString() {
        return consistencyId > 0 ? AppConfigProperties.getTimeFormatted(new Date(consistencyId)) : null;
    }
}

File: src/main/kotlin/eu/getsoftware/hotelico/dto/CustomerDealDto.kt
package eu.getsoftware.hotelico.dto;

import java.util.Date;

import eu.getsoftware.hotelico.utils.ControllerUtils;

class CustomerDealDto: BasicDTO
{
    /**
     * owner activity
     */
    var activityId = 0L

    var consistencyId = 0L

    var timeValid = true

    var lastMinute = false

    var closed = false

    var hotelId: Long? = 0L

    var senderId: Long? = 0L

    var dealDaysDuration = 0

    var totalMoney: Double? = null

    var validFrom: Date? = null

    var validTo: Date? = null

    var title = ""

    private var timeString = ""

    var activityArea = ""

    var dealCode = ""

    var tablePosition = ""

    var dealStatus = ""

    var shortDescription = ""

    var description = ""

    private var pictureUrl: String? = null

    private var previewPictureUrl: String? = null

    constructor(): super()

    fun isActiveBoolean(): Boolean {
        return active
    }

    fun getValidToString(): String {
        return ControllerUtils.dateFormat.format(validTo)
    }

    fun getValidFromString(): String {
        return ControllerUtils.dateFormat.format(validFrom)
    }

    fun getPictureUrl(): String? {
        return ControllerUtils.addHostPrefixOnDemand(pictureUrl)
    }

    fun setPictureUrl(pictureUrl: String) {
        this.pictureUrl = pictureUrl
    }

    fun getPreviewPictureUrl(): String? {
        
        val picUrl = if (ControllerUtils.isEmptyString(previewPictureUrl)) ControllerUtils.PREVIEW_HOTEL_NOT_AVAILABLE_URL else previewPictureUrl

        return ControllerUtils.addHostPrefixOnDemand(picUrl)
    }

    fun setPreviewPictureUrl(previewPictureUrl: String) {
        this.previewPictureUrl = previewPictureUrl
    }

    fun getTimeString(): String? {
        return if (consistencyId > 0) ControllerUtils.getTimeFormatted(Date(consistencyId)) else null
    }
}
